functions/HttpTriggerAPIUsersIdTaskId/__init__.py

```diff
@@ -81,38 +138,45 @@ def connect():
         db_username = os.environ["ENV_DATABASE_USERNAME"]
         db_password = os.environ["ENV_DATABASE_PASSWORD"]
         driver = '{ODBC Driver 17 for SQL Server}'
+        logging.info('Uses hardcoded ODBC Driver 17 for SQL Server string')
+        #assigns the connection string
         conn_string = "Driver={};Server={};Database={};Uid={};Pwd={};Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;".format(
         driver, db_server, db_name, db_username, db_password)
-
-# when passwords are permitted be saved to the config of the Azure function app, 
-# using a single connection string value, saved to the config of the app, 
-# is more economical : <conn_string = os.getenv('SQL_CONNECTION_STRING')>, get the string
-# from "Connection strings" menu on the left, in the database of interest        
+     
         #creates and returns connection variable
-        cnxn = pyodbc.connect(conn_string)
+        logging.debug('Attempting DB connection')
+        try:
+            cnxn = pyodbc.connect(conn_string)
+        except (pyodbc.DatabaseError, pyodbc.InterfaceError) as e:
+            logging.error('Failed to connect to DB: ' + e.args[0])
+            logging.error("Error: " + e.args[1])
+            return func.HttpResponse(status_code=500)
+        logging.debug("Connection to DB successful!")
         return cnxn
-    except Exception:
-        print ("Unable to connect to db. Code 503: Service unavailable")
+    except Exception as e:
+        logging.error('Failure: ' + str(e))
+        return func.HttpResponse(status_code=500)
 
 #MAIN FUNCTION
 def main(req: func.HttpRequest) -> func.HttpResponse:
-    logging.info('Python HTTP trigger function processed a request.')
+    logging.info('Python HTTP trigger function started processing a userId taskId request.')
 
 #collects parameters passed in url        
-    userId = req.params.get('userId')
-    taskId = req.params.get('taskId')
+    userId = req.route_params.get('userId')
+    taskId = req.route_params.get('taskId')
     logging.info('Trying to get userId and taskId')
+    req_body = {}
+    try:
+        req_body = req.get_json()
+    except ValueError:
+        pass
+
     if (not userId) and (not taskId):
-        logging.info("Got neither userId nor taskId")
-        try:
-            req_body = req.get_json()
-        except ValueError:
-            pass
-        else:
-            userId = req_body.get('userId')
-            taskId = req_body.get('taskId')
+        logging.debug("Got neither userId nor taskId")
+        userId = req_body.get('userId')
+        taskId = req_body.get('taskId')

```

Comment:
What's happening here? What's the reason for looking for the route params in the request body? Is it for recovery / diagnosis if the request was formatted incorrectly?

Reply:
yes, this part seems unnecessary. I think this part might be leftovers from before.

Reply:
Per Microsoft documentation, they are recommending to first look for params in the url, and if not available, in the request body:
[httpTrigger](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=python)
[HttpRequest class](https://docs.microsoft.com/en-us/python/api/azure-functions/azure.functions.httprequest?view=azure-python)
With that, the script first looks for route params and if it finds none, it examines the request body.

Reply:
yeah but with this format ("users/{userId:int?}/task/{taskId:int?}") that you got in your function.json, it's almost impossible not to get params, if I'm not mistaken.

Reply:
Should this be added to the list of questions for Toddy, or do you feel strongly enough to go ahead and delete this section? It is prefaced by "if" and will not execute when we get at least one param, I think it does not hurt to leave it in.

Reply:
removed